File: backend/models/url.go
package models

import "time"

type URL struct {
	ID      uint   `gorm:"primaryKey" json:"id"`
	Address string `gorm:"unique;not null" json:"address"`
	Status  string `gorm:"default:queued" json:"status"` // queued, running, done, error

	HTMLVersion string `json:"html_version,omitempty"`
	Title       string `json:"title,omitempty"`

	H1Count int `json:"h1_count,omitempty"`
	H2Count int `json:"h2_count,omitempty"`
	H3Count int `json:"h3_count,omitempty"`
	H4Count int `json:"h4_count,omitempty"`
	H5Count int `json:"h5_count,omitempty"`
	H6Count int `json:"h6_count,omitempty"`

	InternalLinks   int          `json:"internal_links,omitempty"`
	ExternalLinks   int          `json:"external_links,omitempty"`
	BrokenLinks     int          `json:"broken_links,omitempty"`
	BrokenLinksList []BrokenLink `json:"broken_links_details" gorm:"foreignKey:URLID"`

	HasLoginForm bool `json:"has_login_form,omitempty"`

	CreatedAt time.Time `json:"created_at"`
	UpdatedAt time.Time `json:"updated_at"`
}

File: frontend/frontend/src/types/url.ts
export type UrlStatus = 'queued' | 'running' | 'done' | 'error';

export interface BrokenLink {
  id: number;
  url_id: number;
  url: string;
  status_code: number;
  created_at: string;
}

export interface UrlItem {
  id: number;
  address: string;
  status: UrlStatus;
  html_version?: string;
  title?: string;
  h1_count?: number;
  h2_count?: number;
  h3_count?: number;
  h4_count?: number;
  h5_count?: number;
  h6_count?: number;
  internal_links?: number;
  external_links?: number;
  broken_links?: number;
  has_login_form?: boolean;
  broken_links_details?: BrokenLink[];
  created_at: string;
  updated_at: string;
}
